src/library/scanner/libraryscanner.cpp

```diff
@@ -402,10 +402,10 @@ void LibraryScanner::queueTask(ScannerTask* pTask) {
     m_pool.start(pTask);
 }
 
-void LibraryScanner::directoryHashed(const QString& directoryPath,
-                                     bool newDirectory, int hash) {
-    ScopedTimer timer("LibraryScanner::directoryHashed");
-    // qDebug() << "LibraryScanner::directoryHashed" << directoryPath
+void LibraryScanner::directoryHashedAndScanned(const QString& directoryPath,
```

Comment:
This function is the issue. It assumed that a directory must have been scanned as well once it was hashed.